### DISCUSSION TITLE
Parser only returns first result

### DISCUSSION THREAD

**Discussion by MichaelAggerholm** [orchestral/parser#110] [Q&A]
Hello.

I am trying to parse through an xml file:
```
$xml = XmlParser::load('files/books.xml');
    $book = $xml->parse([
    'supplierId' => ['uses' => 'book::id'],
    'author' => ['uses' => 'book.author'],
    'title' => ['uses' => 'book.title'],
    'genre' => ['uses' => 'book.genre'],
    'price' => ['uses' => 'book.price'],
    'publishDate' => ['uses' => 'book.publish_date'],
    'description' => ['uses' => 'book.description'],
]);
```

var_dump($book);

When done, i dump the book data, but i only get the first child. so if i have 100 books , its only the first book that i receive.
Should i be using a foreach or something here? from what i can read, it should parse through the entire file.
Result:

`array(7) { ["supplierId"]=> string(5) "bk101" ["author"]=> string(20) "Gambardella, Matthew" ["title"]=> string(21) "XML Developer's Guide" ["genre"]=> string(8) "Computer" ["price"]=> string(5) "44.95" ["publishDate"]=> string(10) "2000-10-01" ["description"]=> string(63) "An in-depth look at creating applications with XML." }`

**Comment by tego101** [comment]
Did you ever figure this out? It's wild.

**Reply by vbatalov** [reply]
You find fix?

**Reply by tego101** [reply]
no

**Reply by vbatalov** [reply]
try it
```
        $executiveAuthority = $xml->parse([
            ['uses' => 'executiveAuthority.element[::id>element_id,::name>element_name,subject::id>subject_id,subject::name>subject_name,landType::id>landType_id,landType::name>landType_name]'],
        ]);
```

its work for

```
<element id="21" name="Министерство природных ресурсов и экологии Чувашской Республики">
			<ctс:subject id="21" name="Чувашская республика"/>
			<ctс:landType id="1" name="Земли лесного фонда"/>
		</element>
```
